Source/items.cpp

```diff
@@ -4519,79 +4519,69 @@ void SpawnPremium(int pnum)
 void SpawnWitch(int lvl)
 {
 	constexpr int PinnedItemCount = 3;
+	constexpr std::array<int, PinnedItemCount> PinnedItemTypes = { IDI_MANA, IDI_FULLMANA, IDI_PORTAL };

```

Comment:
`GetItemAttrs` taking `int` instead of `_item_indexes` as the second param is not ideal, short of refactoring that function to take an ItemData refence I'd prefer to cast a member of this array to `int` explicitly when it is used than do the cast implicitly when declaring the array.

```suggestion
	constexpr std::array<_item_indexes, PinnedItemCount> PinnedItemTypes = { IDI_MANA, IDI_FULLMANA, IDI_PORTAL };
```

Same for PinnedBookTypes (even though IDI_BOOK2/8/14/20 haven't been defined) and the use in SpawnHealer

Reply:
I can add a commit that implements all item types across the code base or do it as a follow up PR.

Reply:
Eh, might as well do it properly* in a follow up.

*for a given definition of properly :D